File: Algorithms/155_MinStack.java
import java.util.Stack;
class MinStack {
    private Stack<Integer> data;
    private Stack<Integer> min;
    public MinStack() {
        data = new Stack<Integer>();
        min = new Stack<Integer>();
    }
    public void push(int x) {
        data.push(x);
        if (!min.empty() && x > min.peek())
            ;
        else 
            min.push(x); 
    }

    public void pop() {
        int temp = data.peek();
        data.pop();
        if (temp == min.peek())
            min.pop();
    }

    public int top() {
        return data.peek(); 
    }

    public int getMin() {
        return min.peek();        
    }
}

File: Algorithms/cpp/155.MinStack.cpp
class MinStack {
private:
    stack<int> data;
    stack<int> min;
public:
    void push(int x) {
        data.push(x);
        if (min.size() > 0 && x > min.top())
            ;
        else
            min.push(x);
    }

    void pop() {
        int temp = data.top();
        data.pop();
        if (temp == min.top())
            min.pop();
    }

    int top() {
       return data.top(); 
    }

    int getMin() {
       return min.top(); 
    }
};
